### PR TITLE
Convert ERI_script to python

### PR DESCRIPTION
**Pull request by jgfouca**
Other changes:
1) Further reduce possibility of Case object creation in system_test objects
and for preview_namelists. We want as much of the system as possible working
with the same case object (reduces a big potential source of bugs).
2) Remove unneeded year_string tool
3) create_clone library function now does everything, including case_setup
4) Add support for multi-case situations to system_tests_common, you can
select the currently active case.

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes:

User interface changes?: No

Code review: @jedwards4b

### CODE REVIEW

File: utils/python/CIME/SystemTests/system_tests_common.py [hunk 1]
```diff
@@ -58,9 +56,19 @@ def clean_build(self, comps=None):
     def run(self):
         return self._run()
 
-    def _run(self, suffix="base"):
+    def _set_active_case(self, case):
```
Comment (jgfouca) [inline utils/python/CIME/SystemTests/system_tests_common.py]: Key change. Allows a system_test to use multiple different cases, but only 1 at a time.

**Comment by jgfouca** [comment]
@jedwards4b let me know if you need more annotations

File: utils/python/CIME/SystemTests/eri.py [hunk 2]
```diff
@@ -2,16 +2,221 @@
...
+        orig_casevar = orig_case.get_value("CASE")
+
+        #
+        # determine run lengths needed below
+        #
+        stop_n = self._case.get_value("STOP_N")
+        stop_option = self._case.get_value("STOP_OPTION")
+        run_startdate = self._case.get_value("RUN_STARTDATE")
+
+        stop_n1 = stop_n / 6
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: can these be moved to config_tests.xml?

File: utils/python/CIME/SystemTests/eri.py [hunk 3]
```diff
@@ -2,16 +2,221 @@
...
+
+        #
+        # (1) Test run:
+        # do an initial ref1 case run
+        # cloned the case and running there
+        # (NOTE: short term archiving is on)
+        #
+
+        os.chdir(clone1_path)
+        self._set_active_case(clone1)
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: good

File: utils/python/CIME/SystemTests/eri.py [hunk 4]
```diff
@@ -2,16 +2,221 @@
...
+
+        logger.info("ref1 startup: doing a %s %s startup run from %s and 00000 seconds" % (stop_n1, stop_option, start_1))
+        logger.info("  writing restarts at %s %s" % (rest_n1, stop_option))
+        logger.info("  short term archiving is on ")
+
+        clone1.set_value("CONTINUE_RUN", False)
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: If self active case is clone1 why do you use clone1 here instead of self?  Is it the same thing?

File: utils/python/CIME/SystemTests/eri.py [hunk 5]
```diff
@@ -2,16 +2,221 @@
...
+        clone2.set_value("REST_N",        rest_n2)
+        clone2.set_value("HIST_OPTION",   stop_option)
+        clone2.set_value("HIST_N",        hist_n)
+
+        rundir = clone2.get_value("RUNDIR")
+        dout_sr2 = clone2.get_value("DOUT_S_ROOT")
+
+        if not os.path.exists(rundir):
+            os.makedirs(rundir)
+        # lazy, use shell even though we don't have to
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: It took as much to write the comment as it would to do it in python.

File: utils/python/CIME/SystemTests/eri.py [hunk 6]
```diff
@@ -2,16 +2,221 @@
...
+        self._case.set_value("REST_N"        , rest_n3)
+        self._case.set_value("HIST_OPTION"   , stop_option)
+        self._case.set_value("HIST_N"        , stop_n2)
+        self._case.set_value("DOUT_S"        , False)
+
+        rundir = self._case.get_value("RUNDIR")
+        if not os.path.exists(rundir):
+            os.makedirs(rundir)
+        # lazy, use shell even though we don't have to
+        rest_path = os.path.join(dout_sr2, "rest", "%s-%s" % (refdate_3, ref_to_d3))
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: move repeated code into a subroutine?

File: utils/python/CIME/case_run.py [hunk 7]
```diff
@@ -245,35 +245,31 @@ def DoDataAssimilation(case, da_script, lid):
...
-            if data_assimilation:
-                DoDataAssimilation(case, data_assimilation_script, lid)
-
-        resubmitCheck(case)
-    except:
-        logger.warning("Exception in case_run: %s" % sys.exc_info()[1])
-        return False
-    else:
-        return True
+    # Set up the run, run the model, do the postrun steps
```
Comment (jedwards4b) [inline utils/python/CIME/case_run.py]: Why this change?

**Comment by jedwards4b** [comment]
This looks great.   A few minor nits.

File: utils/python/CIME/SystemTests/eri.py (hunk 2 above)
Comment (jgfouca) [inline utils/python/CIME/SystemTests/eri.py]: I thought that system wasn't fully functional yet?

File: utils/python/CIME/SystemTests/eri.py (hunk 4 above)
Comment (jgfouca) [inline utils/python/CIME/SystemTests/eri.py]: Yes, same thing.

File: utils/python/CIME/SystemTests/eri.py (hunk 5 above)
Comment (jgfouca) [inline utils/python/CIME/SystemTests/eri.py]: Done. Changed to python.

File: utils/python/CIME/SystemTests/eri.py (hunk 6 above)
Comment (jgfouca) [inline utils/python/CIME/SystemTests/eri.py]: Done.

File: utils/python/CIME/case_run.py (hunk 7 above)
Comment (jgfouca) [inline utils/python/CIME/case_run.py]: I wanted to let except failures escape, this ensures non-zero error code without having to manage a bool all the way up the call chain.

File: utils/python/CIME/SystemTests/eri.py (hunk 2 above)
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/eri.py]: I thought you were taking care of that :-)  no worries.

**Comment by jedwards4b** [comment]
You marked a couple of things as done that aren't updated here - I assume they are done in your ready to merge sandbox?   merge away...

**Comment by jgfouca** [comment]
Yeah, coded them up. Will push and merge.
